### ISSUE TITLE
Option to turn auto-updating of axes ranges off/on

### ISSUE THREAD

**Issue by kasuteru** [bokeh/bokeh#8171]
**tl;dr: An option to turn auto-updating of axes ranges on/off in a figure would make plotting more transparent and versatile.**

This was initially supposed to be a bug report similar to #7281, since I had the same intentions: I change a CDS, then manually set _x_range_ and _y_range_. After searching here, I found that my actual issue was that after setting the axes manually by doing 
```
         p.x_range.start = dict_default_zoom[new][0][0]
         p.x_range.end   = dict_default_zoom[new][0][1]
         p.y_range.start = dict_default_zoom[new][1][0]
         p.y_range.end   = dict_default_zoom[new][1][1]
```
, I was immediately updating another CDS in the same plot - and this automatically resets the axes ranges again, as explained in [this stackoverflow question](https://stackoverflow.com/questions/42513690/python-bokeh-how-to-turn-off-auto-update-of-axes). The solution from the same question that also worked for me is kind of curious: You have to set _x_range_ and _y_range_ when creating the figure, this prevents automatically fitting to new data, like this: `f = figure(x_range=[0, 10], y_range=[0, 100])`

The problem I see here is that 
a) _this behavior is documented nowhere_ (as far as I have seen) and kind of unintuitive - I spent all morning on looking for what I was doing wrong before realizing what the problem was. I can write a PR that adds a short explanation of this behavior to the docs if this is wanted.
b) _one cannot change the behavior after the fact_ - either the plot auto-updates every time, or never. In my use case, it would be nice to auto-update for new, unknown data and stay fixed for data where I already entered the optimal ranges into a table.

I haven't looked at the bokeh source much until now, but I imagine this would be fairly easy to implement? My guess would be that there is already a flag somewhere that turns auto-updating off/on, and it simply would need to be exposed?

**Comment by bryevdv** [comment]
@kasuteru There should not be any new work necessary, setting `start` / `end` on a default `DataRange1d` is supposed to permanently override auto-ranging. Are you saying it does not? If that is the case, that is a bug, so please provide a *complete, minimal script* to reproduce the issue.

**Comment by kasuteru** [comment]
@bryevdv Thanks for your answer. 

> An explicitly supplied range end. If provided, will override automatically computed end value.
This sentence was not entirely clear to me. If I understand you correctly, the following is how it should work:

1. If I set `end` manually for e.g. x_range, autoscaling will be disabled.
1. If I then set `end` manually to `None`, autoscaling will be enabled again.
1. If I manually zoom into my figure in bokeh server, autoscaling is still enabled, but only when my data changes. At least, this is the behaviour I observed: I zoom into an autoscaled figure (which, to my understanding, has to save the current x_range value _somewhere_...), then I interactively changed the CDS, which triggered a rescaling of my axes.

Assuming the above is the intended behavior, we can close this I think.
